File: 000-099/026-Remove-Duplicates-from-Sorted-Array.cpp
// Time complexity: O(n).
// Space complexity: O(1) extra memory.

class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int i = 0;
        for (auto num : nums) {
            if (i < 1 || nums[i - 1] != num) {
                nums[i++] = num;
            }
        }
        return i;
    }
};

File: 000-099/026-Remove-Duplicates-from-Sorted-Array.java
class Solution {
    public int removeDuplicates(int[] nums) {
        if (nums == null)
            throw new NullPointerException("Array is null!");
        
        int i = 0;
        for (int j = 1; j < nums.length; j++) {
            if (nums[j] != nums[i]) {
                i++;
                nums[i] = nums[j];
            }
        }
        return i + 1;
    }
}